### DISCUSSION TITLE
ERR_CONNECTION_TIMED_OUT upon restore of app with PVC

### DISCUSSION THREAD

**Discussion by jc0705** [vmware-tanzu/velero#3261] [Community support Q&A] [answered]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)
- got a test app (wordpress) with PVC running
- Got a s3 database running (type :ingress)
- Configured "restic-restore-action-config" configmap to pull from internal repo
- Installed velero with the following code: 
`$ velero install --bucket velero --provider aws --secret-file credentials-velero --use-volume-snapshots=false --use-restic --backup-location-config region=minio,s3ForcePathStyle="true",s3Url=$S3_URL,publicUrl=$S3_IP --image $INTERNAL_REPO/velero/velero:v1.4.2 --plugins $INTERNAL_REPO/velero/velero-plugin-for-aws:latest`
- Modified restic DaemonSet
- Annotations done for the backup-volumes
- Backup created and restored the namespace successfully

After the back up (pods and pvc) is restored successfully and running, I tried to access the wordpress app at the newly given ip address ( xx.xxx.xx.11) generated by the loadbalancer but was redirected to xx.xxx.xx.31 (the external ip of original wordpress service) with ERR_CONNECTION_TIMED_OUT.

**What did you expect to happen:**
The restored wordpress app and can be accessed with the newly assigned IP address.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

Original service before back up:
```
$ kubectl get svc -A
NAMESPACE     NAME              TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)         AGE
wordpress     wordpress         LoadBalancer   xx.xx.xxx.88    xx.xxx.xx.31   80:30678/TCP    2d17h
wordpress     wordpress-mysql   ClusterIP      None             <none>        3306/TCP        2d17h
```
Restored service:
```
$ kubectl get svc -A
NAMESPACE     NAME              TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)         AGE
wordpress     wordpress         LoadBalancer   xx.xx.xxx.141   xx.xxx.xx.11   80:31519/TCP    37m
wordpress     wordpress-mysql   ClusterIP      None             <none>        3306/TCP        37m
```
**EDIT:** I tried doing backups with and without annotations for the backup volumes... 
A _CLEAN_ app is restored successfully and is reachable at the new external-ip address given for backup without annotation.. While an annotated backup gives me the error `ERR_CONNECTION_TIMED_OUT` at the new external-ip address.

**Environment:**

- Velero version (use `velero version`): 
```
Client:
        Version: v1.4.2
        Git commit: 56a08a4d695d893f0863f697c2f926e27d70c0c5
Server:
        Version: v1.4.2
```
- Velero features (use `velero client config get features`): 
`features: <NOT SET>`
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7", GitCommit:"be3d344ed06bff7a4fc60656200a93c74f31f9a4", GitTreeState:"clean", BuildDate:"2020-02-11T19:34:02Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7+vmware.1", GitCommit:"7be6fd53a1c9885bf9f470673f1c14280aab6386", GitTreeState:"clean", BuildDate:"2020-02-13T23:24:07Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`):
```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
```

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by ashish-amarnath** [comment, marked as answer]
@jc0705 From the outset this doesn't seem like an issue with velero considering
> EDIT: I tried doing backups with and without annotations for the backup volumes...
A CLEAN app is restored successfully and is reachable at the new external-ip address given for backup without annotation.. While an annotated backup gives me the error ERR_CONNECTION_TIMED_OUT at the new external-ip address.

Can you also please tell us what it is that you are backing up and what's the data that is getting backed up from the volumes?

**Reply by carlisia** [reply]
This discussion has been moved to an issue: https://github.com/vmware-tanzu/velero/issues/3349.

**Comment by jc0705** [comment]
@ashish-amarnath Hi i was doing a backup on wordpress example from [kubernetes.io](https://kubernetes.io/docs/tutorials/stateful-application/mysql-wordpress-persistent-volume/). I suspect the IP address is stored in the MySQL PV that came along with the app, which caused the redirect of the ip address upon restoration (not really sure). 

But as for now I'm able to backup and restore another app with it's PV so I guess the issue lies with the content we are backing up.
